### PR TITLE
frost-radio

### LINKED ISSUE #237: Redesign radio-button-group and remove nearestOfType usage
yield radio button group as contextual component instead of nearestOfType
  {{#radio-group as |controls|}}
    {{controls.radio-button}}
{{/radio-group}}

### PR DESCRIPTION
**Pull request by dafortin**
#fix#

#Fix #237

# CHANGELOG
* **Rework** radio button and radio group
* **Update** documentation
* **Update** tests
* **Update** demo

### CODE REVIEW

**Comment by rox163** [comment]
could you update the PR description with the Changelog header so at least the changes are tracked in there.

File: addon/components/frost-radio-button.js [hunk 1]
```diff
@@ -51,32 +58,27 @@ export default Component.extend(PropTypeMixin, {
...
    * @param {String} value - radio button's value
+   * @param {String} receivedHook - hook received from parent
    * @returns {String} the concatenated hook name
    */
-  hook (value) {
-    const radioGroupHook = get(this, 'parentView.hook')
-    if (radioGroupHook) {
-      return `${radioGroupHook}-button-${value}`
-    }
+  hook (value, receivedHook) {
```
Comment (rox163) [inline addon/components/frost-radio-button.js]: could we also set a hookQualifier property [set to 'value'] here so we stay consistent with the qualifier usage across the addon?

**Review by rox163 — CHANGES_REQUESTED** [review]
nicely done

File: addon/components/frost-radio-button.js (hunk 1 above)
Comment (dafortin) [reply addon/components/frost-radio-button.js]: good idea

**Review by rox163 — APPROVED** [review]

**Comment by rox163** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/282/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**